pkg/ccip/chainaccessor/ton_accessor.go

```diff
@@ -2,37 +2,63 @@ package chainaccessor
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/binary"
 	"errors"
+	"fmt"
 	"time"
 
+	"github.com/xssnick/tonutils-go/address"
+	"github.com/xssnick/tonutils-go/tlb"
 	"github.com/xssnick/tonutils-go/ton"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
 
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/logutil"
+
+	"github.com/smartcontractkit/chainlink-ton/pkg/ccip/bindings/onramp"
+	"github.com/smartcontractkit/chainlink-ton/pkg/ccip/codec"
 	"github.com/smartcontractkit/chainlink-ton/pkg/logpoller"
+	"github.com/smartcontractkit/chainlink-ton/pkg/logpoller/types/cellquery"
+	"github.com/smartcontractkit/chainlink-ton/pkg/ton/hash"
 )
 
 type TONAccessor struct {
-	lggr      logger.Logger
-	client    ton.APIClientWrapped
-	logPoller logpoller.LogPoller
+	lggr          logger.Logger
+	chainSelector cciptypes.ChainSelector
+	client        ton.APIClientWrapped
+	logPoller     logpoller.LogPoller
+	bindings      map[string]*address.Address
+	addrCodec     codec.AddressCodec
 }
 
 var _ ccipocr3.ChainAccessor = (*TONAccessor)(nil)
 
 func NewTONAccessor(
 	lggr logger.Logger,
+	chainSelector cciptypes.ChainSelector,
 	client ton.APIClientWrapped,
 	logPoller logpoller.LogPoller,
 	addrCodec ccipocr3.AddressCodec,
 ) (ccipocr3.ChainAccessor, error) {
 	// TODO: validate state of client and logPoller (should be initialized in NewChain)
+	if client == nil {
```

Comment:
@ogtownsend We would have to think about the design here, the caller ([oracle creator](https://github.com/smartcontractkit/chainlink/blob/262522919fc85cb957200894a70f060e3ffdddd9/core/capabilities/ccip/oraclecreator/plugin.go#L454)) doesn't have access to rpc client here and we don't want to introduce chain specific code in the plugin.go

Reply:
nvm, this should be handled by relayer/ccipprovider